okapi-core/src/main/java/org/folio/okapi/util/CorsHelper.java

```diff
@@ -29,57 +30,60 @@ private CorsHelper() {
    *
    * @param router - {@link Router}
    */
-  public static void addCorsHandler(Router router) {
+  public static void addCorsHandler(Router router, final TenantManager tenantManager) {
 
+    // set delegate CORS for special cases
     router.routeWithRegex("^/_/invoke/tenant/[^/ ]+/.*").handler(ctx -> {
-      ctx.data().put(CHECK_DELEGATE_CORS, true);
-      ctx.next();
+      String origPath = ctx.request().path();
+      String tenantId = origPath.replaceFirst("^/_/invoke/tenant/([^/ ]+)/.*$", "$1");
+      String newPath = origPath.replaceFirst("^/_/invoke/tenant/[^/ ]+(/.*$)", "$1");
```

Comment:
```
    router.routeWithRegex("^/_/invoke/tenant/([^/ ]+)(/.*)").handler(ctx -> {
      String tenantId = ctx.pathParam("param0");
      String newPath = ctx.pathParam("param1");
```
https://vertx.io/docs/vertx-web/java/#_capturing_path_parameters_with_regular_expressions

Reply:
Nice! Thank you.